src/frontend/src/pages/FinancePage/ReimbursementRequestDetailPage/ReimbursementRequestDetailsView.tsx

```diff
@@ -20,16 +20,34 @@ import LocalShippingIcon from '@mui/icons-material/LocalShipping';
 import ConfirmationNumberIcon from '@mui/icons-material/ConfirmationNumber';
 import CheckIcon from '@mui/icons-material/Check';
 import { isReimbursementRequestApproved } from '../../../utils/reimbursement-request.utils';
+import AddSABONumberModal from './AddSABONumberModal';
+import { useState } from 'react';
+import { useSetSaboNumber } from '../../../hooks/finance.hooks';
+import { useToast } from '../../../hooks/toasts.hooks';
 
 interface ReimbursementRequestDetailsViewProps {
   reimbursementRequest: ReimbursementRequest;
 }
 
 const ReimbursementRequestDetailsView: React.FC<ReimbursementRequestDetailsViewProps> = ({ reimbursementRequest }) => {
   const theme = useTheme();
+  const toast = useToast();
   const totalCostBackgroundColor = theme.palette.mode === 'dark' ? theme.palette.grey[800] : theme.palette.grey[200];
   const user = useCurrentUser();
   const history = useHistory();
+  const [addSaboNumberModalShow, setAddSaboNumberModalShow] = useState<boolean>(false);
+  const { mutateAsync: setSaboNumber } = useSetSaboNumber(reimbursementRequest.reimbursementRequestId);
+
+  const onSaboFormSubmit = async (data: { saboNumber: number }) => {
+    try {
+      await setSaboNumber(data);
+    } catch (error: unknown) {
+      if (error instanceof Error) {
+        toast.error(error.message);
+      }
+    }
+    setAddSaboNumberModalShow(false);
+  };
```

Comment:
Move this method and the hook call, and the toast hook into the new component you made, and just pass the component a setOpen method and a reimbursementRequestId. This way all the logic for setting the sabo Id is contained within the component, since there's already a lot going on in this file